.github/workflows/lint.yml

```diff
@@ -42,6 +42,13 @@ jobs:
           # date with target branch before merging, anyway.
           # See https://github.com/shadow/shadow/issues/2166
           ref: ${{ github.event.pull_request.head.sha }}
+      # rustfmt fails if referenced modules aren't present, so we need
+      # to generate the bindings.
+      - name: Generate bindings
+        run: |
+          sudo apt-get install -y cmake
+          cmake -S . -B build
+          (cd src && cargo check)
```

Comment:
> You need to at least run cmake. You don't need to compile it, though we don't expose a way to just run cmake without compiling in the setup script...

Correction, you currently need to run cmake to generate shd-config.h *and* run at least `cargo check` to actually generate the bindings

Reply:
Maybe another option is to add a command like `setup configure` that runs `cmake` and `cargo check` without actually building. I think it'd mostly only be useful in the CI though; users will generally just want to do a build anyway

Reply:
> One alternative is that we could keep checking in the generated bindings for now, even though we regenerate them on every build. That'd solve this issue, and let us notice changes in the generated bindings, while I think still having most of the benefits of this PR. e.g. it'd still remove the problem of forgetting to rebuild the bindings, and we could probably still safely disable the bindings lint. Wdyt?

This would be okay with me. Maybe we'd want to still run the lint for the case where someone makes a small change and commits it without rebuilding (the lint has caught me doing this many times).

> Maybe another option is to add a command like setup configure that runs cmake and cargo check without actually building. I think it'd mostly only be useful in the CI though; users will generally just want to do a build anyway

This sounds okay too, but yeah it probably wouldn't be a very popular option.

One other solution (at least so that the project has all valid code that you could run rustfmt on), could be the following:

`src/main/lib.rs`:

```rust
/// cbindgen:ignore
pub mod cshadow {
    include!("cshadow.rs");
}
```

This way there is always a valid module, even if the file doesn't exist. (And the same with the log bindings file.)

But any of these (or none at all) is fine with me. We can always change it later if we want.

Reply:
> > Does this cause any issues for other tools like rust analyzer, cargo doc, etc before you've built shadow?
>
> I haven't hit this issue with other tools, no.

One thing I wanted to check out was the effect on rust-analyzer if you have the automatic running of build scripts disabled, but I'm using Ale with vim and it doesn't support rust-analyzer config options, so I can't disable `rust-analyzer.cargo.buildScripts.enable` to try that. But we can figure that out in the future if we need to.

Reply:
Ah, using `include!` looks like a nice simple solution. That's actually what they do in the bindgen docs, but they didn't explain why.